ArduPlane/control_modes.cpp

```diff
@@ -149,13 +149,39 @@ void Plane::read_control_switch()
 uint8_t Plane::readSwitch(void) const
 {
     uint16_t pulsewidth = RC_Channels::get_radio_in(g.flight_mode_channel - 1);
-    if (pulsewidth <= 900 || pulsewidth >= 2200) return 255;            // This is an error condition
-    if (pulsewidth <= 1230) return 0;
-    if (pulsewidth <= 1360) return 1;
-    if (pulsewidth <= 1490) return 2;
-    if (pulsewidth <= 1620) return 3;
-    if (pulsewidth <= 1749) return 4;              // Software Manual
-    return 5;                                                           // Hardware Manual
+    if (g.extra_modes == 0) {
+        // Old ranges, for backwards compatibility.
+        if (pulsewidth <= 900 || pulsewidth >= 2200) return 255;            // This is an error condition
+        if (pulsewidth <= 1230) return 0;
+        if (pulsewidth <= 1360) return 1;
+        if (pulsewidth <= 1490) return 2;
+        if (pulsewidth <= 1620) return 3;
+        if (pulsewidth <= 1749) return 4;              // Software Manual
+        return 5;                                                           // Hardware Manual
+    } else {
+        // New ranges.
+        if (pulsewidth <= 900 || pulsewidth >= 2200) return 255;            // This is an error condition
```

Comment:
this max bounds check is redundant. Do it once before the extra_modes check

Reply:
Good call, thank you, done.